Introduce delete file modal

jseminck opened the pull request:
This PR adds a modal for confirming file deletion, instead of using `alert()`. If you're happy with it, we can also do something similar for deleting sandboxes.

I'm not particularly happy with the design. It's not something I'm very good at. I tried using elements that exist on other places already. Actually, the red colour for the Delete button doesn't feel right...

I'm not sure if you want this. Maybe you want to keep `alert()` but I assumed it was done this way initially because it was the easiest approach. 😄 

![deletemodal](https://user-images.githubusercontent.com/9586897/29753826-4d2a97d6-8b82-11e7-96a8-16ac1666cb5c.gif)

As usual, I learned quite a lot by looking at your Modal implementation (1 instance of `Modal` and keeping everything else in the state), and I'll probably use the same approach for another project where we recently started adding more modals. 

**PS:** I noticed my diff includes a lot of trailing comma removals. This was done by the pre-commit hook, is it expected?

CompuIves commented:
Nice @jseminck, this is indeed something we desperately needed! I like this implementation, but think it would be even better if we have a general 'Alert' component that we can use on all places we use alert. A solution to the buttons is to make the right button blue (primary color) and the left one yellow (secondary color).

jseminck commented:
Great idea. I'll tackle it tonight, re-use it for delete sandbox and post new screenshots 👍

jseminck commented:
Sorry! I have been a bit busy. I looked for the secondary colour the other day but I couldn't find it. Did you mean the same yellow when you hover over one of the current primary buttons?

The secondary colour inside the theme object is also some shade of blue.

That yellow would look a bit odd as a secondary cancel button because it has the same colour as hovering over the primary one. So when you hover over primary, both buttons have the same colour.

lbogdan commented:
Maybe something like this? (from [here](https://ux.stackexchange.com/questions/49991/should-yes-delete-it-be-red-or-green/))
![](https://i.stack.imgur.com/RdF94.png)

CompuIves commented:
Good points @jseminck, and that looks beautiful @lbogdan! Though I don't think we should go with a red primary button, since we a simpler UX/UI if we use the same type of modal on the whole website (without exceptions). A red primary button only makes sense for deletion modals.

Maybe we could do something like this (quick mockup):

![screen shot 2017-09-02 at 13 05 13](https://user-images.githubusercontent.com/587016/29995031-6e97fe16-8fdf-11e7-83f3-2c17ea15fff9.png)

lbogdan commented:
Oh, I thought we were only talking about *destructive* modals. :smile:

jseminck commented:
Updated: changed button styles, renamed component and moved it to /components/ folder, also used Alert for sandbox deletion. 

Maybe the hover for cancel still needs some adjustment 😄

jseminck commented on src/app/pages/Sandbox/Editor/Workspace/SandboxActions/index.js:
```diff
@@ -28,18 +24,43 @@ const PrivacySelect = styled.select`
+          onCancel={modalActions.closeModal}
+          onDelete={async () => {
+            await this.props.deleteSandbox(this.props.id);
+            await this.props.newSandboxUrl();
```
Do these function calls really need to `await`? Right now the modal wouldn't be closed until `deleteSandbox()` and `newSandboxUrl()` are completed. - not sure if that is intended...

CompuIves replied:
Hmm, yes, seems like we can remove the `await` statementes.

CompuIves commented:
Super great work! Nice!

CompuIves commented:
Nice, works perfectly! Merging it in now...
